Source/HTTP/httpcall.cpp

```diff
@@ -137,6 +162,105 @@ HRESULT CALLBACK HC_CALL::PerfomAsyncProvider(XAsyncOp op, XAsyncProviderData co
     }
 }
 
+#if !HC_NOZLIB
+
+#if HC_PLATFORM == HC_PLATFORM_WIN32 || HC_PLATFORM == HC_PLATFORM_GDK
+
+void CALLBACK HC_CALL::CompressRequestBody(void* c, bool canceled)
+{
+    PerformContext* context{ static_cast<PerformContext*>(c) };
+    HC_CALL* call{ context->call };
+
+    if (canceled)
+    {
+        XAsyncComplete(context->asyncBlock, E_ABORT, 0);
+        return;
+    }
+
+    if (call->traceCall)
+    {
+        HC_TRACE_INFORMATION(HTTPCLIENT, "HC_CALL::CompressRequestBody [ID %llu] Iteration %d", TO_ULL(call->id), call->m_iterationNumber);
+    }
+
+    HCHttpCallRequestBodyReadFunction clientRequestBodyReadCallback{ nullptr };
+    size_t requestBodySize{};
+    void* clientRequestBodyReadCallbackContext{ nullptr };
+    HRESULT hr = HCHttpCallRequestGetRequestBodyReadFunction(call, &clientRequestBodyReadCallback, &requestBodySize, &clientRequestBodyReadCallbackContext);
+
+    if (FAILED(hr) || !clientRequestBodyReadCallback)
+    {
+        HC_TRACE_ERROR(HTTPCLIENT, "HC_CALL::CompressRequestBody: Unable to get client's RequestBodyRead callback");
+        return;
+    }
+
+    size_t uncompressedRequestyBodySize = 0;
+    const size_t bufferSize = 500;
+    uint8_t* uncompressedRequestyBodyBuffer = new uint8_t[requestBodySize];
```

Comment:
@jplafonta is there a better way to define the buffer we'll be reading from the client? 

Since this is a dynamically defined value, not sure how we can achieve this on the stack, so I've allocated it on the heap and deleting it after compression function.

Reply:
You can't avoid using a dynamic allocation since we don't know the body size ahead of time. I'd use a Vector for a few reasons:

1) Its easier to resize on the fly
2) It automatically uses the client's allocators - you're using 'new' here which won't
3) you don't have to worry about freeing the allocation manually

Reply:
Fixed